src/core/symbology/classification/qgsclassificationmethod.h

```diff
@@ -0,0 +1,290 @@
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSCLASSIFICATIONMETHOD_H
+#define QGSCLASSIFICATIONMETHOD_H
+
+#include "qgis_core.h"
+#include "qgsprocessingparameters.h"
+
+class QgsVectorLayer;
+class QgsRendererRange;
+
+
+#ifdef SIP_RUN
+// This is required for the ConvertToSubClassCode to work properly
+// so RTTI for casting is available in the whole module.
+% ModuleCode
+#include "qgsclassificationequalinterval.h"
+#include "qgsclassificationjenks.h"
+#include "qgsclassificationprettybreaks.h"
+#include "qgsclassificationquantile.h"
+#include "qgsclassificationstandarddeviation.h"
+% End
+#endif
+
+
+
+/**
+ * \ingroup core
+ * QgsClassificationRange contains the information about a classification range
+ */
+class CORE_EXPORT QgsClassificationRange
+{
+  public:
+    //! Constructor
+    QgsClassificationRange( const QString &label, const double &lowerBound, const double &upperBound )
+      : mLabel( label )
+      , mLowerBound( lowerBound )
+      , mUpperBound( upperBound )
+    {}
+    //! Returns the lower bound
+    double lowerBound() const {return mLowerBound;}
+    //! Returns the upper bound
+    double upperBound() const {return mUpperBound;}
+
+    //! Returns the lower bound
+    QString label() const {return mLabel;}
+
+  private:
+    QString mLabel;
+    double mLowerBound;
+    double mUpperBound;
+};
+
+
+
+/**
+ * \ingroup core
+ * QgsClassification is an abstract class for implementations of classification methods
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsClassificationMethod SIP_ABSTRACT
+{
+
+#ifdef SIP_RUN
+    SIP_CONVERT_TO_SUBCLASS_CODE
+    if ( dynamic_cast<QgsClassificationEqualInterval *>( sipCpp ) )
+      sipType = sipType_QgsClassificationEqualInterval;
+    else if ( dynamic_cast<QgsClassificationJenks *>( sipCpp ) )
+      sipType = sipType_QgsClassificationJenks;
+    else if ( dynamic_cast<QgsClassificationPrettyBreaks *>( sipCpp ) )
+      sipType = sipType_QgsClassificationPrettyBreaks;
+    else if ( dynamic_cast<QgsClassificationQuantile *>( sipCpp ) )
+      sipType = sipType_QgsClassificationQuantile;
+    else if ( dynamic_cast<QgsClassificationStandardDeviation *>( sipCpp ) )
+      sipType = sipType_QgsClassificationStandardDeviation;
+    else
+      sipType = 0;
+    SIP_END
+#endif
+
+  public:
+
+    //! Defines the class position
+    enum ClassPosition
+    {
+      LowerBound,
+      Inner,
+      UpperBound
+    };
+
+    /**
+      * Creates a classification method.
+      * \param valuesRequired if TRUE, this means that the method requires a set of values to determine the classes
```

Comment:
instead of the separate bool arguments here, I'd rather see a flags approach. It'll be more readable and more extensible.